### PR TITLE
Fix transmutation with micromamba

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Closes https://github.com/conda/constructor/issues/674.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Review by jezdez — APPROVED** [review]

**Comment by jaimergp** [comment]
I looked a bit into this and fixed, I think:

- Problems started somewhere between Apr 3rd and 21st (see [this PR](https://github.com/conda/constructor/pull/660) for the first occurrence). It used to work before, but there were no significant releases in that time that explain the problem.
- The problem is not about extraction this time, but hash verification. Basically the extracted archives will have a different hash in their `info/repodata_record.json` if they have been transmuted, and micromamba doesn't like that, so it thinks that the cache is corrupted and tries to download the artifact again. The solution is to remove the md5 for those files so micromamba doesn't check. 
- To summarize, the URL and md5 in the explicit input file must match the ones present in the repodata_record.json on the corresponding info/ directory. Otherwise it constitutes a cache mismatch.

**Review by jezdez — APPROVED** [review]
